lib/battle/CBattleInfoCallback.cpp

```diff
@@ -610,12 +607,11 @@ bool CBattleInfoCallback::battleCanShoot(const battle::Unit * attacker, BattleHe
 	if(attacker->creatureIndex() == CreatureID::CATAPULT && defender) //catapult cannot attack creatures
 		return false;
 
-	if(attacker->canShoot()
+	return attacker->canShoot()
 		&& battleMatchOwner(attacker, defender)
 		&& defender->alive()
-		&& (!battleIsUnitBlocked(attacker) || attacker->hasBonusOfType(Bonus::FREE_SHOOTING)))
-		return true;
-	return false;
+		&& (!battleIsUnitBlocked(attacker)
+		|| attacker->hasBonusOfType(Bonus::FREE_SHOOTING));

```

Comment:
Let's say honest I'd rather to have more lines here instead of this kind of stuff.
It's not like your change really improved this code readability.

Reply:
Let's say fair.
When you have such code
`if(statement) return true;
else return false;
`
it's nonsense keeping it seperate.  Better write `return statement;`
Such things should make code more simple to read, and if they don't, then something is wrong with code itself. It's good way to show that something need refactor and it's not expressive enought.

Reply:
It's just matter of taste, but for me this is more readable:
```
if(statement1)
    return false;
if(statement2)
    return false;
if(statement3)
    return false;
return true;
```
I just generally dont care about vertical space used, but I certainly not going to pretend like it's really any importance for me. As long as it's are done right (e.g with fixes you already applied) I have nothing against changes like that.

Though I wish to know what @alexvins think since there changes in his code. He can merge it if it's fine by him.

Reply:
Hmm, as for me it's also not so important, I just looked around the code and changed it by the way :).

Reply:
Though it's always funny to think about the consequences of such a "small" change.

Reply:
I`m OK with this particular change, but I will oppose merging statements above to this condition.

Reply:
Yep, agree that simplify to many is bad.